File: backend/src/main/scala/no/hamre/polet/parser/FileDownloader.scala
package no.hamre.polet.parser

import no.hamre.polet.util.Slf4jLogger
import scala.io.Source

trait FileDownloader {
  def download(url: String): String
}

class FileDownloaderImpl(encoding: String) extends FileDownloader with Slf4jLogger {

  override def download(url: String): String = {
    log.info(s"Prepare to download from $url")
    val html = Source.fromURL(url, encoding)
    val s = html.mkString
    log.debug(s)
    s
  }


}

File: backend/src/main/kotlin/no/hamre/polet/parser/FileDownloader.kt
package no.hamre.polet.parser

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import java.net.URL
import java.nio.charset.Charset

interface FileDownloader {
  fun download(url: String): String
}

@Component
class FileDownloaderImpl(@Value("\${vinmonopolet.encoding}") private val encoding: String) : FileDownloader {
  private val log = LoggerFactory.getLogger(this.javaClass)

  override fun download(url: String): String {
    log.info("Prepare to download from $url")
    val html = URL(url).readText(Charset.forName(encoding))
    log.debug(html)
    return html
  }


}
